Sources/PubNub/Networking/Routers/PublishRouter.swift

```diff
@@ -15,11 +15,33 @@ import Foundation
 struct PublishRouter: HTTPRouter {
   // Nested Endpoint
   enum Endpoint: CustomStringConvertible {
-    case publish(message: AnyJSON, channel: String, shouldStore: Bool?, ttl: Int?, meta: AnyJSON?)
-    case compressedPublish(message: AnyJSON, channel: String, shouldStore: Bool?, ttl: Int?, meta: AnyJSON?)
-    case fire(message: AnyJSON, channel: String, meta: AnyJSON?)
-    case signal(message: AnyJSON, channel: String)
-    case file(message: FilePublishPayload, shouldStore: Bool?, ttl: Int?, meta: AnyJSON?)
+    case publish(
+      message: AnyJSON, channel: String,
+      customMessageType: String?, shouldStore: Bool?,
+      ttl: Int?, meta: AnyJSON?
+    )
+    case compressedPublish(
+      message: AnyJSON, channel: String,
+      customMessageType: String?, shouldStore: Bool?,
+      ttl: Int?, meta: AnyJSON?
+    )
+    case fire(
+      message: AnyJSON,
+      channel: String,
+      meta: AnyJSON?
+    )
+    case signal(
+      message: AnyJSON,
+      channel: String,
+      customMessageType: String?
+    )
+    case file(
+      message: FilePublishPayload,
+      customMessageType: String?,
```

Comment:
Indeed, there's no mention about it in specs. I asked Smitha and here's the response I got. It works as you assumed:


> The `/v1/files/publish-file/...`  endpoint translates to a publish endpoint in the server so that `custom_message_type` will work exactly like the `type` parameter. No change